### PR TITLE
Upgrade rust to 1.2.0

### PR DESCRIPTION
**Pull request by EPashkin**

### CODE REVIEW

File: mingw-w64-rust/PKGBUILD [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 _realname=rust
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
-pkgver=1.1.0
-pkgrel=3
+pkgver=1.2.0
+pkgrel=4
```
Comment (Alexpux) [inline mingw-w64-rust/PKGBUILD]: With new version you need set ````pkgrel```` to ````1````. It increasing only for packages with the same version.

**Comment by EPashkin** [comment]
Fixed `pkgrel=1`.
Pass `release-channel=stable` to configure.

**Comment by EPashkin** [comment]
When building got many errors about llvm
```
I:/msys64/MINGW-packages/mingw-w64-rust/src/rustc-1.2.0/src/rustllvm/RustWrapper.cpp:248:48: error: no matching function for call to 'llvm::DIVariable::DIVariable(llvm::MDNode*)'
     return DIT(ref ? unwrap<MDNode>(ref) : NULL);
                                                ^
In file included from I:/msys64/MINGW-packages/mingw-w64-rust/src/rustc-1.2.0/src/llvm/include/llvm/IR/DebugInfo.h:25:0,
                 from I:/msys64/MINGW-packages/mingw-w64-rust/src/rustc-1.2.0/src/rustllvm/rustllvm.h:55,
                 from I:/msys64/MINGW-packages/mingw-w64-rust/src/rustc-1.2.0/src/rustllvm/RustWrapper.cpp:11:
I:/msys64/MINGW-packages/mingw-w64-rust/src/rustc-1.2.0/src/llvm/include/llvm/IR/DebugInfoMetadata.h:1735:3: note: candidate: llvm::DIVariable::DIVariable(llvm::LLVMContext&, unsigned int, llvm::Metadata::StorageType, unsigned int, unsigned int, llvm::ArrayRef<llvm::Metadata*>)
   DIVariable(LLVMContext &C, unsigned ID, StorageType Storage, unsigned Tag,
   ^
I:/msys64/MINGW-packages/mingw-w64-rust/src/rustc-1.2.0/src/llvm/include/llvm/IR/DebugInfoMetadata.h:1735:3: note:   candidate expects 6 arguments, 1 provided
I:/msys64/MINGW-packages/mingw-w64-rust/src/rustc-1.2.0/src/rustllvm/RustWrapper.cpp: In instantiation of 'DIT unwrapDI(LLVMMetadataRef) [with DIT = llvm::DICompositeType; LLVMMetadataRef = LLVMOpaqueMetadata*]':
I:/msys64/MINGW-packages/mingw-w64-rust/src/rustc-1.2.0/src/rustllvm/RustWrapper.cpp:740:66:   required from here
I:/msys64/MINGW-packages/mingw-w64-rust/src/rustc-1.2.0/src/rustllvm/RustWrapper.cpp:248:48: error: no matching function for call to 'llvm::DICompositeType::DICompositeType(llvm::MDNode*)'
     return DIT(ref ? unwrap<MDNode>(ref) : NULL);
                                                ^
In file included from I:/msys64/MINGW-packages/mingw-w64-rust/src/rustc-1.2.0/src/llvm/include/llvm/IR/DebugInfo.h:25:0,
                 from I:/msys64/MINGW-packages/mingw-w64-rust/src/rustc-1.2.0/src/rustllvm/rustllvm.h:55,
                 from I:/msys64/MINGW-packages/mingw-w64-rust/src/rustc-1.2.0/src/rustllvm/RustWrapper.cpp:11:
I:/msys64/MINGW-packages/mingw-w64-rust/src/rustc-1.2.0/src/llvm/include/llvm/IR/DebugInfoMetadata.h:840:3: note: candidate: llvm::DICompositeType::DICompositeType(llvm::LLVMContext&, llvm::Metadata::StorageType, unsigned int, unsigned int, unsigned int, uint64_t, uint64_t, uint64_t, unsigned int, llvm::ArrayRef<llvm::Metadata*>)
   DICompositeType(LLVMContext &C, StorageType Storage, unsigned Tag,
   ^
I:/msys64/MINGW-packages/mingw-w64-rust/src/rustc-1.2.0/src/llvm/include/llvm/IR/DebugInfoMetadata.h:840:3: note:   candidate expects 10 arguments, 1 provided
/MINGW-packages/mingw-w64-rust/src/rustc-1.2.0/mk/rustllvm.mk:59: recipe for target 'x86_64-pc-windows-gnu/rustllvm/RustWrapper.o' failed
```

**Comment by martell** [comment]
It seems to use its own version of llvm

https://github.com/rust-lang/llvm/tree/874dc4ee4cb782056469f003831bcda3e4cdf0df

You will probably have to wait until 3.7.0 is tagged and w release packages
for that.
Then we can patch rust to ignore its own version and use the installed one
via pacman.

**Comment by EPashkin** [comment]
Thanks, martell.
rust-git building fine by tag 1.2.0 (except that it contains libgcc_s_seh-1.dll and libstdc++-6.dll that already exists in my system, currently I fix it by commenting lines in `MINGW-packages\mingw-w64-rust-git\src\rust\src\etc\make-win-dist.py`).
So I wait for 21 August for LLVM v3.7.0.

**Comment by Alexpux** [comment]
@EPashkin, here we remove gcc dl's that rust want to distribute after install: https://github.com/Alexpux/MINGW-packages/blob/master/mingw-w64-rust/PKGBUILD#L53-L54

**Comment by EPashkin** [comment]
@Alexpux, oh, thanks.
I remember that it was in rust's PKGBUIILD but for unknown reason don't see it yourself :(

**Comment by EPashkin** [comment]
Bad, with llvm 3.7.0.also not builds (after fix version check in configure).
Seems only way build with included llvm (without --llvm-root=)
```
I:\msys64\mingw64\bin\llc.exe: I:/msys64/MINGW-packages/mingw-w64-rust/src/rustc-1.2.0/src/rt/rust_try.ll:27:29: error: expected instruction opcode
    landingpad { i8*, i32 } personality i8* bitcast (i32 (...)* @rust_eh_personality_catch to i8*)
                            ^
/MINGW-packages/mingw-w64-rust/src/rustc-1.2.0/mk/rt.mk:95: recipe for target 'x86_64-pc-windows-gnu/rt/rust_try.o' failed
```

**Pull request merged**